extended/src/main/java/apoc/vectordb/VectorDb.java

```diff
@@ -0,0 +1,234 @@
+        Object id = conf.isAllResults() ? m.get(conf.getIdKey()) : null;
+        List<Double> embedding = hasEmbedding ? (List<Double>) m.get(conf.getVectorKey()) : null;
+        Map<String, Object> metadata = hasMetadata ? (Map<String, Object>) m.get(conf.getMetadataKey()) : null;
+        // in case of get operation, e.g. http://localhost:52798/collections/{coll_name}/points with Qdrant db,
+        // score is not present
+        Double score = Util.toDouble(m.get(conf.getScoreKey()));
+        String text = conf.isAllResults() ? (String) m.get(conf.getTextKey()) : null;
+
+        Entity entity = handleMapping(tx, db, mapping, metadata, embedding);
+        if (entity != null) entity = Util.rebind(tx, entity);
+        return new EmbeddingResult(id, score, embedding, metadata, text, entity);
+    }
+
+    private static Entity handleMapping(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metadata, List<Double> embedding) {
+        if (mapping.getProp() == null) {
+            return null;
+        }
+        if (MapUtils.isEmpty(metadata)) {
+            throw new RuntimeException("To use mapping config, the metadata should not be empty. Make sure you execute `YIELD metadata` on the procedure");
+        }
+        Map<String, Object> metaProps = new HashMap<>(metadata);
+        if (mapping.getLabel() != null) {
+            return handleMappingNode(tx, db, mapping, metaProps, embedding);
+        } else if (mapping.getType() != null) {
+            return handleMappingRel(tx, db, mapping, metaProps, embedding);
+        } else {
+            throw new RuntimeException("Mapping conf has to contain either label or type key");
+        }
+    }
+
+    private static Entity handleMappingNode(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metaProps, List<Double> embedding) {
+        String query = "CREATE CONSTRAINT IF NOT EXISTS FOR (n:%s) REQUIRE n.%s IS UNIQUE"
+                .formatted(mapping.getLabel(), mapping.getProp());
+        db.executeTransactionally(query);
+
+        try {
+            Node node;
+            try (Transaction transaction = db.beginTx()) {
+                Object propValue = metaProps.remove(mapping.getId());
+                node = transaction.findNode(Label.label(mapping.getLabel()), mapping.getProp(), propValue);
+                if (node == null && mapping.isCreate()) {
+                    node = transaction.createNode(Label.label(mapping.getLabel()));
+                    node.setProperty(mapping.getProp(), propValue);
+                }
+                if (node != null) {
+                    setProperties(node, metaProps);
+                }
+                transaction.commit();
+            }
+
+            String setVectorQuery = "CALL db.create.setNodeVectorProperty($entity, $key, $vector)";
```

Comment:
we can set the property to a float array ourselves, no need to call cypher here.